ext/LogDensityProblemsADForwardDiffExt.jl

```diff
@@ -18,49 +18,95 @@ end
 # Load DiffResults helpers
 include("DiffResults_helpers.jl")
 
-struct ForwardDiffLogDensity{L, C} <: ADGradientWrapper
+struct ForwardDiffLogDensity{L, C <: ForwardDiff.Chunk, T <: Union{Nothing,ForwardDiff.Tag},
+                             G <: Union{Nothing,ForwardDiff.GradientConfig}} <: ADGradientWrapper
+    "supports zero-order evaluation `logdensity(ℓ, x)`"
     ℓ::L
-    gradientconfig::C
+    "chunk size for ForwardDiff"
+    chunk::C
+    "tag, or `nothing` for the default"
+    tag::T
+    "gradient config, or `nothing` if created for each evaluation"
+    gradient_config::G
 end
 
 function Base.show(io::IO, ℓ::ForwardDiffLogDensity)
     print(io, "ForwardDiff AD wrapper for ", ℓ.ℓ,
-          ", w/ chunk size ", length(ℓ.gradientconfig.seeds))
+          ", w/ chunk size ", ForwardDiff.chunksize(ℓ.chunk))
 end
 
 _chunk(chunk::ForwardDiff.Chunk) = chunk
 _chunk(chunk::Integer) = ForwardDiff.Chunk(chunk)
 
 _default_chunk(ℓ) = _chunk(dimension(ℓ))
 
-_default_gradientconfig(ℓ, chunk, ::Nothing) = _default_gradientconfig(ℓ, chunk, zeros(dimension(ℓ)))
-function _default_gradientconfig(ℓ, chunk, x::AbstractVector)
-    return ForwardDiff.GradientConfig(Base.Fix1(logdensity, ℓ), x, _chunk(chunk))
+function Base.copy(fℓ::ForwardDiffLogDensity{L,C,T,<:ForwardDiff.GradientConfig}) where {L,C,T}
+    @unpack ℓ, chunk, tag, gradient_config = fℓ
+    ForwardDiffLogDensity(ℓ, chunk, tag, copy(gradient_config))
 end
 
 """
-    ADgradient(:ForwardDiff, ℓ; x, chunk, gradientconfig)
-    ADgradient(Val(:ForwardDiff), ℓ; x, chunk, gradientconfig)
+$(SIGNATURES)
+
+Make a `ForwardDiff.GradientConfig` for input `x`. `tag = nothing` generates the default tag.
+
+Return the function for evaluating log density (with a vector argument) as the second value.
+"""
+function _make_gradient_config(x, ℓ, chunk, tag)
+    ℓ′ = Base.Fix1(logdensity, ℓ)
+    c = _chunk(chunk)
+    gradient_config = if tag ≡ nothing
+        ForwardDiff.GradientConfig(ℓ′, x, c)
+    else
+        ForwardDiff.GradientConfig(ℓ′, x, c, tag)
+    end
+    gradient_config, ℓ′
+end
```

Comment:
IMO it would be a bit simpler (see below) to directly pass `Base.Fix1(logdensity, ℓ)` to this function.
````suggestion
Make a `ForwardDiff.GradientConfig` for function `f` and input `x`. `tag = nothing` generates the default tag.
```
function _make_gradient_config(f::F, x, chunk, tag) where {F}
    c = _chunk(chunk)
    gradient_config = if tag ≡ nothing
        ForwardDiff.GradientConfig(f, x, c)
    else
        ForwardDiff.GradientConfig(f, x, c, tag)
    end
    gradient_config
end
````